sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubClient.cs

```diff
@@ -270,7 +282,103 @@ public virtual EventSender CreateSender(SenderOptions senderOptions = default)
         ///
         [EditorBrowsable(EditorBrowsableState.Never)]
         public override string ToString() => base.ToString();
+
+        /// <summary>
+        ///   Performs the actions needed to validate the <see cref="EventHubClientOptions" /> associated
+        ///   with this client.
+        /// </summary>
+        ///
+        /// <param name="clientOptions">The set of options to validate.</param>
+        ///
+        /// <remarks>
+        ///   In the case that the options violate an invariant or otherwise represent a combination that
+        ///   is not permissible, an appropriate exception will be thrown.
+        /// </remarks>
+        ///
+        protected virtual void ValidateClientOptions(EventHubClientOptions clientOptions)
+        {
+            // If there were no options passed, they cannot be in an invalid state.
+
+            if (clientOptions == null)
+            {
+                return;
+            }
+
+            // A proxy is only valid when websockets is used as the transport.
+
+            if ((clientOptions.ConnectionType == ConnectionType.AmqpTcp) && (clientOptions.Proxy != null))
+            {
+                throw new ArgumentException(String.Format(CultureInfo.CurrentCulture, Resources.ProxyMustUseWebsockets), nameof(clientOptions));
+            }
+        }
+
+        /// <summary>
+        ///   Performs the actions needed to validate the <see cref="ConnectionStringProperties" /> associated
+        ///   with this client.
+        /// </summary>
+        ///
+        /// <param name="properties">The set of properties parsed from the connection string associated this client.</param>
+        /// <param name="connectionStringArgumentName">The name of the argument associated with the connection string; to be used when raising <see cref="ArgumentException" /> variants.</param>
+        ///
+        /// <remarks>
+        ///   In the case that the prioperties violate an invariant or otherwise represent a combination that
+        ///   is not permissible, an appropriate exception will be thrown.
+        /// </remarks>
+        ///
+        protected virtual void ValidateConnectionStringProperties(ConnectionStringProperties properties,
```

Comment:
Does this have to be protected?

Reply:
In support of unit testing, yes, I believe there is value in allowing mocks to inherit and accept a potentially invalid set of options.

Reply:
internal virtual with InternalsVisibleTo works in this case.

Reply:
The intent is to allow consumers to mock the behavior of the client as well as our own efforts.  Making it internal supports our own testing, but limits what external callers can do.

Reply:
Having an empty protected constructor would allow customers to avoid going through validation.

Reply:
Revisited notes and scenarios around testing and decided that I'm in agreement with your thoughts;  I've made connection string parsing internal and allowed for child classes to set the client options directly, which gives the flexibility needed to avoid the connection string constructors.